### ISSUE TITLE
[FG:InPlacePodVerticalScaling] Add /resize subresource to request pod resource resizing

### ISSUE THREAD

**Issue by vinaykul** [kubernetes/kubernetes#109553]
### What would you like to be added?

In-place Pod vertical scaling [feature](/kubernetes/kubernetes/pull/102884/) currently relies on PATCH verb applied to Pod objects. This issue tracks a task to design and implement /resize subresource for Pods as opposed to PATCH, add RBAC controls to it, and additionally consider supporting it for Pod superset objects such as Jobs, RS, Deploys etc.

This came up during KEP discussions with Tim Hockin as we revisited first version of the API.
Please see the following discussion threads for context:
https://github.com/kubernetes/enhancements/pull/1883#discussion_r459156714
https://github.com/kubernetes/enhancements/pull/1883#discussion_r464627665
https://github.com/kubernetes/enhancements/pull/1883#discussion_r448752659
https://github.com/kubernetes/enhancements/pull/1883#discussion_r555668219
https://github.com/kubernetes/enhancements/pull/1883#issuecomment-814273899

### Why is this needed?

It may be cleaner and more desirable to use /resize subresource to request a desired resize, and such a resource can be optionally supported on objects such as Deployments, Jobs etc.

**Comment by tallclair** [comment]
@iholder101 the discussion is still relevant, but I don't know if the design is committed. If you want to propose an addition to the KEP, that's probably the next step.

/reopen
/remove-lifecycle rotten

**Comment by iholder101** [comment]
> @iholder101 the discussion is still relevant, but I don't know if the design is committed. If you want to propose an addition to the KEP, that's probably the next step.
> 
> /reopen /remove-lifecycle rotten

I see it's already mentioned in the KEP, both under the proposal [[1](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/1287-in-place-update-pod-resources#subresource)] and under the graduation criteria for beta [[2](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/1287-in-place-update-pod-resources#beta)].

If it still has priority and is considered a graduation criteria for Beta, I'd be happy to start working on it. If priority drops or it's no longer relevant, please let me know.

/assign

[1] https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/1287-in-place-update-pod-resources#subresource
[2] https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/1287-in-place-update-pod-resources#beta

**Comment by vinaykul** [comment]
@iholder101 This is still relevant, and a fun extension to work on. It is a better way than PATCH verb, and VPA should use it. Please update the KEP as you make progress on it.

**Comment by tallclair** [comment]
I still need to read through the full context for this, but if it's just a matter of having finer grained RBAC control than full pod update permissions, then I wonder if a ValidatingAdmissionPolicy would be sufficient for the VPA case instead of a new subresource. The discussion on https://github.com/kubernetes/enhancements/pull/3617 is also relevant.

Either way, is this really a beta blocker?

**Comment by vinaykul** [comment]
Let me find some time to read through @lavalamp KEP. The reason I marked it as beta blocker is the API becomes official at beta IIRC (we will be committed to supporting PATCH for resize). If that's not the case, then it need not block beta but still is nice to have.

**Comment by tallclair** [comment]
@iholder101 probably best to hold off on this until we agree on the shape of the API in https://github.com/kubernetes/enhancements/pull/4704

**Comment by iholder101** [comment]
> @iholder101 probably best to hold off on this until we agree on the shape of the API in [kubernetes/enhancements#4704](https://github.com/kubernetes/enhancements/pull/4704)

Thanks for the update!
/hold

**Comment by tallclair** [comment]
I'm reassigning this to @AnishShah to unblock progress. @iholder101 please let me know ASAP if you were still hoping to take this.

/assign @AnishShah 
/unassign @iholder101
